rules/list-item.js

```diff
@@ -102,12 +102,24 @@ function validateList(list, file) {
 			continue;
 		}
 
-		const [link, ...description] = paragraph.children;
-
-		if (link.type === 'text') {
+		if (paragraph.children[0].type === 'text') {
 			continue;
 		}
 
+		let [link, ...description] = paragraph.children;
+
+		// Might have children like {image} {text} {link} - {descrition}
+		// Keep discarding elements until we find a link
+		//                                       or linkReference
+		if (!/link/.test(link.type)) {
+			for (let i = 0; i < description.length - 1; i++) {
```

Comment:
Use a `for-of` loop.

Reply:
worse, I don't think I need tow be testing link's type twice. while loop is probably better?